Server/src/main/java/org/gluu/oxauth/model/config/WebKeysConfiguration.java

```diff
@@ -6,17 +6,58 @@
 
 package org.gluu.oxauth.model.config;
 
-import javax.enterprise.inject.Vetoed;
-
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import org.gluu.oxauth.model.configuration.AppConfiguration;
 import org.gluu.oxauth.model.configuration.Configuration;
+import org.gluu.oxauth.model.jwk.JSONWebKey;
 import org.gluu.oxauth.model.jwk.JSONWebKeySet;
 
+import javax.enterprise.inject.Vetoed;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * @author Yuriy Movchan
  * @version 03/15/2017
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
 @Vetoed
 public class WebKeysConfiguration extends JSONWebKeySet implements Configuration {
+
+    /**
+     * Configuration needed to get configuration done in LDAP about algorithms supported.
+     */
+    private AppConfiguration appConfiguration;
```

Comment:
We added logic into bean. It works but isn't it better to keep in `JwkRestWebServiceImpl` were it naturally fits? In this case we don't need this field with setter in keys bean.

Reply:
The reason is because I overriten getKeys() method, in `JwkRestWebServiceImpl` we only use `toString()` method. Also anyone that would need to use WebKeysConfiguration will only use the list of keys that should be considered, however if I filter the list in `JwkRestWebServiceImpl` it will be only used for that response.

Reply:
Exactly, it's produce spaghetty : 
1) we can't get loaded keys anymore (only filtered one). 
2) if `setAppConfiguration` is not called (and it's not in many places), we will get NPE.

I vote to keep behavior and data separated. So yes, it should be used only for that response.